salt/modules/peeringdb.py

```diff
@@ -0,0 +1,316 @@
+# -*- coding: utf-8 -*-
+'''
+PeeringDB Module
+================
+
+.. versionadded:: Fluorine
+
+Execution module for the basic interaction with the
+`PeeringDB <https://www.peeringdb.com/>`_ API.
+
+While for GET operations (the functions prefixed by ``get_``) the credentials
+are optional, there are some specific details that are visible only to
+authenticated users. Moreover, the credentials are required when adding or
+updating information. That means, the module can equally work out of the box
+without any further configuration with the limitations imposed by the PeeringDB
+API.
+
+For complete API documentation, please refer to https://www.peeringdb.com/apidocs/.
+
+Configuration (in the opts or Pillar):
+
+.. code-block:: yaml
+
+    peeringdb:
+      username: salt
+      password: 5@1t
+'''
+from __future__ import absolute_import
+
+# Import python libs
+import logging
+log = logging.getLogger(__name__)
+
+# Import salt modules
+import salt.utils.http
+try:
+    from salt.utils import clean_kwargs
+except ImportError:
+    from salt.utils.args import clean_kwargs
+
+__virtualname__ = 'peeringdb'
+__proxyenabled__ = ['*']
+
+PEERINGDB_URL = 'https://www.peeringdb.com/api'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def _get_auth(username=None,
+              password=None):
+    peeringdb_cfg = __salt__['config.merge']('peeringdb', default={})
+    if not username:
+        username = peeringdb_cfg.get('username', username)
+    if not password:
+        password = peeringdb_cfg.get('password', password)
+    return username, password
+
+
+def _build_url(endpoint, id=None):
+    if id:
+        return '{base}/{endp}/{id}'.format(base=PEERINGDB_URL, endp=endpoint, id=id)
+    return '{base}/{endp}'.format(base=PEERINGDB_URL, endp=endpoint)
+
+
+def _get_endpoint(endpoint, id=None, **kwargs):
+    username, password = _get_auth(kwargs.pop('username', None),
+                                   kwargs.pop('password', None))
+    kwargs = clean_kwargs(**kwargs)
+    url = _build_url(endpoint, id=id)
+    ret = {
+        'comment': '',
+        'result': True,
+        'out': None
+    }
+    res = salt.utils.http.query(url,
```

Comment:
This will default to HTTP. Should we prefer HTTPS given that we're passing credentials in the clear here?

Reply:
Definitely. Is there an option to `salt.utils.http.query` for that? I've never used it, I'm afraid. Though, the base URL is `PEERINGDB_URL = 'https://www.peeringdb.com/api'` and I have assumed that it would use that. Let me know and I'll update, thanks!

Reply:
Ah! You are right. Yup, we're good here. False alarm on my part.